server/cache/cache.go

```diff
@@ -10,17 +12,24 @@ import (
 )
 
 type ResourceCache struct {
+	ctx    context.Context
+	client kubernetes.Interface
 	v1.ServiceAccountLister
-	v1.SecretLister
 }
 
 func NewResourceCache(client kubernetes.Interface, ctx context.Context, namespace string) *ResourceCache {
 	informerFactory := informers.NewSharedInformerFactoryWithOptions(client, time.Minute*20, informers.WithNamespace(namespace))
 	cache := &ResourceCache{
+		ctx:                  ctx,
+		client:               client,
 		ServiceAccountLister: informerFactory.Core().V1().ServiceAccounts().Lister(),
-		SecretLister:         informerFactory.Core().V1().Secrets().Lister(),
 	}
 	informerFactory.Start(ctx.Done())
 	informerFactory.WaitForCacheSync(ctx.Done())
 	return cache
 }
+
+func (c *ResourceCache) GetSecret(namespace string, secretName string) (*v12.Secret, error) {
```

Comment:
Reason to introduce this method is if we want to introduce caching in future, we can easily change its implementation and keep external contract same